src/c_api/c_api_common.h

```diff
@@ -91,7 +91,7 @@ struct MXAPIThreadLocalEntry {
     data->resize(shapes.size());
     size_t size = 0;
     for (const auto& s : shapes) {
-      CHECK_GE(s.ndim(), 0);
+      if (s.ndim() > 0);
```

Comment:
what is this?

Reply:
`s.ndim()` can be -1 if the shape is unknown. In that situation, it should not contribute to the calculation of the total size.

Reply:
why not remove this line of code?